### PR TITLE
Feature/index full text

### LINKED ISSUE #135: Full text search for documents.
Display in results view

### LINKED ISSUE #230: Index full text of documents into solr so the text inside of a document can be searched.

### PR DESCRIPTION
**Pull request by straleyb**
Fixes #135, Fixes #230 

This adds a new group of objects that allows for parsing OCR documents into pages and words. This is used by the document model to append the full text of a document to the solr document to add searching capabilities.

### CODE REVIEW

File: app/models/blacklight_config/search_field_configuration.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ def run
     private
 
     def fields
-      [all_field] | normal_fields
+      [full_text] | [all_field] | normal_fields
```
Comment (terrellt) [inline app/models/blacklight_config/search_field_configuration.rb]: [full_text, all_field] | normal_fields

File: app/models/document.rb [hunk 2]
```diff
@@ -1,3 +1,21 @@
 class Document < GenericAsset
   has_derivatives :pdf_pages, :ocr, :thumbnail
+
+  def to_solr
```
Comment (terrellt) [inline app/models/document.rb]: We should have this stuff as part of the indexer.

**Comment by straleyb** [comment]
This should be good to go now.

**Comment by terrellt** [comment]
@straleyb https://coveralls.io/builds/3306805/source?filename=lib%2Foregon_digital%2Fderivatives%2Fprocessors%2Focr_processor.rb

**Comment by straleyb** [comment]
@terrellt Should I add a test to check that the ocr processor correctly displays an error when it fails pdftotext?

**Comment by straleyb** [comment]
Alright coverage is fixed.

File: app/models/document.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
 class Document < GenericAsset
   has_derivatives :pdf_pages, :ocr, :thumbnail
+
```
Comment (jechols) [inline app/models/document.rb]: You know, newlines don't grow on trees!  Reduce, reuse, recycle!

**Pull request merged**